static/src/javascripts/projects/common/modules/discussion/comment-count.js

```diff
@@ -0,0 +1,141 @@
+    const elements = context.querySelectorAll(`[${ATTRIBUTE_NAME}]`);
+
+    return [
+        ...elements,
+    ].reduce((groupedVals: Object, el: HTMLElement): Object => {
+        const attrVal = el.getAttribute(ATTRIBUTE_NAME);
+
+        if (!groupedVals[attrVal]) {
+            groupedVals[attrVal] = [];
+        }
+
+        groupedVals[attrVal].push(el);
+
+        return groupedVals;
+    }, {});
+};
+
+const getContentIds = (indexedElements: Object): string =>
+    Object.keys(indexedElements).sort().join(',');
+
+const getContentUrl = (el: HTMLElement): string => {
+    const a = el.getElementsByTagName('a')[0];
+
+    return `${a ? a.pathname : ''}#comments`;
+};
+
+const updateElement = (el: HTMLElement, count: number): Function => (): Promise<
+    void
+> => {
+    const url = el.getAttribute('data-discussion-url') || getContentUrl(el);
+
+    if (el.getAttribute('data-discussion-closed') === 'true' && count === 0) {
+        // Discussion is closed and had no comments, we don't want to show a comment count
+        return Promise.resolve();
+    }
+
+    const format = el.getAttribute('data-commentcount-format') || '';
+    const html = getTemplate(
+        {
+            url,
+            icon: inlineSvg('commentCount16icon', ['inline-tone-fill']),
+            count: integerCommas(count) || '',
+        },
+        format
+    );
+    const meta = el.getElementsByClassName('js-item__meta');
+    const containers = meta.length ? [...meta] : [el];
+
+    return fastdom.write(() => {
+        containers.forEach(container => {
+            container.insertAdjacentHTML('beforeend', html);
+        });
+        el.removeAttribute(ATTRIBUTE_NAME);
+        el.classList.remove('u-h');
+    });
+};
+
+const renderCounts = (
+    counts: Array<{ id: string, count: number }>,
+    indexedElements: IndexedElements
+): void => {
+    const elementUpdatesGrouped = counts.map(c =>
+        indexedElements[c.id].map(el => updateElement(el, c.count))
+    );
+
+    const elementUpdatesFlattened = elementUpdatesGrouped.reduce(
+        (a, b) => a.concat(b),
+        []
+    );
+
+    Promise.all(
+        elementUpdatesFlattened.map(elementUpdate => elementUpdate())
+    ).then(() => {
+        mediator.emit('modules:commentcount:loaded', counts);
+    });
+};
+
+const getCommentCounts = (context: HTMLElement): void => {
+    fastdom.read(() => {
+        const indexedElements = getElementsIndexedById(
+            context || document.body
+        );
+        const endpoint = `${COUNT_URL}${getContentIds(indexedElements)}`;
+
+        fetchJSON(endpoint, {
+            mode: 'cors',
+        }).then(response => {
+            if (response && response.counts) {
+                renderCounts(response.counts, indexedElements);
+            }
+        });
+    });
+};
+
+const init = (): void => {
+    if (document.body && document.body.querySelector('[data-discussion-id]')) {
```

Comment:
Could this be simplified to

```js
if(document.querySelector(`[${ATTRIBUTE_NAME}]`)) {
  getCommentCounts(document.body);
}
```

Edit: Oh, there is the expensive selector again. Any chance of caching the result/ isolate the read in a function, which then could cache?

Edit2: `fastdom.read()` around it?

?